src/hypothesis/searchstrategy/types.py

```diff
@@ -129,6 +135,19 @@ def from_typing_type(thing):
 for t in integer_types:
     _global_type_lookup[t] = st.integers()
 
+if hasattr(os, 'PathLike'):  # pragma: no branch
+    _global_type_lookup[os.PathLike] = \
+        st.fspaths(allow_pathlike=False).map(st._PathLike)
+
+# Also check for PY3 because there is a Python 2 pathlib port on pypi
+# which wont work
+if PY3 and pathlib is not None:  # pragma: no branch
+    _global_type_lookup[pathlib.Path] = \
```

Comment:
Ah, thanks. I got confused by `from_type(pathlib.PosixPath).example()` already returning a PosixPath and assumed that it fell back to the base class strategy which would be the right thing here. But it just falls back to `builds(PosixPath)`..